BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -1012,22 +1012,17 @@ extension StackLayout {
 
         // First pass, use natural to get a measurement
         let axisItems = items(axisPressure: .natural)
-        // Second pass, inherit mode so that we fill properly
-        let crossItems = items(axisPressure: nil)
+        // Second pass, fill assigned space on axis
+        let crossItems = items(axisPressure: .fill)
 
         let frames = _frames(axisItems: axisItems, crossItems: crossItems, in: vectorConstraint)
         
-        let crossPressure: StrictPressureMode = {
-            switch axis {
-            case .horizontal:
-                return context.mode.vertical
-            case .vertical:
-                return context.mode.horizontal
-            }
-        }()
-        
-        switch (alignment, crossPressure) {
-        case (.fill, .natural):
+        // If we're in fill mode, we'll do an additional pass to ensure we fill the assigned space,
+        // in case:
+        // - cross pressure is natural, so we still need to fill the calculated size
+        // - we skipped a pass during cross segments, and didn't fill along axis yet
+        switch alignment {
+        case .fill:
```

Comment:
oof, this is another hit to Strict SPL performance, right?

Reply:
Yeah. There might be a tighter condition we could apply, but this much fixed the regression tests.